Datto recap blog post (#507)
trinodb/trino.io: _posts/2023-07-25-trino-fest-2023-datto.md

---
layout: post
title: "Starburst Galaxy: A romance of many architectures"
author: "Benjamin Jeter, Ryan Duan"
excerpt_separator: <!--more-->
image: /assets/blog/trino-fest-2023/Datto.png
show_pagenav: false
---

Let's cut straight to the chase with this lightning talk from Benjamin Jeter, a
data architect, platform manager, and data engineer at Datto. For those that are
not familiar with Datto, they are an American cybersecurity and data backup

> company who are the leading global provider of security and cloud-based software

Reviewer: ```suggestion
company. They're the leading global provider of security and cloud-based software
```
This sentence is a little long, let's split it in half.

solutions purpose-built for Managed Service Providers (MSPs). In Benjamin's
talk, he goes through some of the considerations and design goals of a reference

> architecture pattern that they use and why they chose to use Trino on Starburst

Reviewer 2: You cant use Trino on Starburst Galaxy ... Trino is inside it..

chose to to use Starburst Galaxy.

And then in the next paragraph have a quick sentence about it being a SaaS with Trino inside?

Galaxy.

<!--more-->

{% youtube K3AlAWB-Gmg %}

<a class="btn btn-pink btn-md" target="_blank" href="/assets/blog/trino-fest-2023/TrinoFest2023Datto.pdf">
  Check out the slides!
</a>

## Recap

But you might be wondering: what does Ben mean when he says "reference
architecture"? A reference architecture pattern is a pattern for making
arbitrary data available to end users in a reproducible and modular way. It's an
opinionated representation of what best practices look like for a given class of
use cases. You can almost think of it as a conceptual tool for thinking
critically about specific patterns through a pragmatic balance of simplicity and
effectiveness. However, it is not something that will work for every use case
and not necessarily the best solution.

The main design goal that Benjamin had was to facilitate near real-time data

> access, while using only Trino. In addition, he wanted it to be simple, easy to

Reviewer: ```suggestion
access while using only Trino. In addition, he wanted it to be simple, easy to
```

understand, flexible, and adaptable. Accomplishing this design goal requires
many steps, such as first having a daily batch transform that transforms JSON

> into Iceberg and serve as T-1 data. Then he created an unpartitioned external

Reviewer 2: what is T-1 data ?

table that is rebuilt every day as part of the daily batch transform. Using the

> Hive connector with this table allows Datto to have scan on query semantics,

Reviewer 2: There is no Hive connector in Starburst Galaxy ... its Great Lakes connectivity, or S3 catalog or Hive table format with the Great Lakes connectivity .. but actually .. I think he is using Iceberg right?

which enables data access about as real-time as you can get it without a
streaming solutions like Kafka or Kinesis. Benjamin shows how easy it is to

> design a use case with just a couple lines of code using Trino on Starburst

Reviewer 2: reword .. not "on"

Reviewer: "with" is probably better.
